src/control/lib/control/auto_test.go

```diff
@@ -0,0 +1,354 @@
+//
```

Comment:
Good coverage for the helper methods, but I'd like to see one for the top level as well. It needn't exercise everything the helper tests already verified, of course, but you could test things like nil request, function pointers are used if passed in, checkNetwork/checkStorage are called, etc.

Reply:
I'd like to address this in a follow-up when I add Target_count parameter generation if that's okay, I don't feel that the level of coverage is unacceptable for an initial commit

Reply:
Okay with me.